Hymns/Tags/TagStore.swift

```diff
@@ -60,6 +75,21 @@ class TagStoreRealmImpl: TagStore {
             action(favorite)
         }.toNotification()
     }
+
+    // MARK: Realm Query Functions
+
+    /** Can be used either with a value to specificially query for one tag or without the optional to query all tags besides favorites*/
+    func querySelectedTags(tagSelected: String?) -> Results<TagEntity> {
+        guard let specificTag = tagSelected else {
+            return realm.objects(TagEntity.self)
+        }
+        return realm.objects(TagEntity.self).filter(NSPredicate(format: "tags == %@", specificTag))
+    }
+
+    func queryTagsForHymn(hymnIdentifier: HymnIdentifier) -> Results<TagEntity> {
+        let convertedKey = TagEntity.createPrimaryKey(hymnIdentifier: hymnIdentifier, tag: "")
+        return realm.objects(TagEntity.self).filter(NSPredicate(format: "primaryKey CONTAINS[c] %@ AND tags != %@", convertedKey, "_*_favorited_*_"))
```

Comment:
Use the static constant in the FavoritesEntity instead of hardcoded string

Reply:
I guess the question here is also that we could probably just get rid of the second filter for it not being favorites. Since won't all favorites now only be stored in the FavoriteEntity and not the TagEntity. So there won't be any found when we query all object in TagEntity